### PR TITLE
WIP: Touchid before Unlock on iOS

### PR DESCRIPTION
**Pull request by cmgustavo**

### CODE REVIEW

File: src/js/controllers/walletHome.js [hunk 1]
```diff
@@ -335,69 +335,70 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
...
-              $scope.$digest();
+              $timeout(function() {
+                $scope.$digest();
+              });
               return;
             }
 
-            profileService.signTxProposal(txp, function(err, txpsi) {
-              self.setOngoingProcess();
+            profileService.unlockFC(function(err) {
```
Comment (matiu) [inline src/js/controllers/walletHome.js]: we should not call unlock if the wallet is not locked. that will ask a password.

**Comment by matiu** [comment]
Looks good, but we need to test it in details.

I think we should remove the timeout in locking and add:
```
  // for all clients
   if (client.hasPrivateKeyEncrypted()) {
     client.lock();
   }
```
on app startup, so we can lock open wallet in case of a crash.

**Comment by cmgustavo** [comment]
Agree. Thanks for your comment.

I 'll set it as a `WIP` PR until we decide a better solution

**Comment by cmgustavo** [comment]
Updated.

**Pull request merged**